pulsar-client/src/main/java/org/apache/pulsar/client/impl/BinaryProtoLookupService.java

```diff
@@ -177,6 +180,46 @@ public String getServiceUrl() {
         return serviceAddress.toString();
     }
 
+    @Override
+    public CompletableFuture<List<String>> getTopicsUnderNamespace(NamespaceName namespace) {
+        return getTopicsUnderNamespace(serviceAddress, namespace);
+    }
+
+    private CompletableFuture<List<String>> getTopicsUnderNamespace(InetSocketAddress socketAddress,
+                                                                    NamespaceName namespace) {
+        CompletableFuture<List<String>> topicsFuture = new CompletableFuture<List<String>>();
+
+        client.getCnxPool().getConnection(socketAddress).thenAccept(clientCnx -> {
```

Comment:
At the higher level, we should have a way to retry few times this call, otherwise if we're unlike when starting our consumer we might get exception even if there is a brief transient problem (eg: 1 single broker going down).

Reply:
Thanks, will add it.